tests/cpp/test_multidevice_communications.cpp

```diff
@@ -337,11 +359,15 @@ TEST_P(CommunicationTest, Allreduce) {
 }
 
 TEST_P(CommunicationTest, ReduceScatter) {
-  IrContainer container;
-  auto* communication = IrBuilder::createInContainer<Communication>(
-      &container,
+  hir::HostIrContainer container;
+  FusionGuard fg(&container);
+  auto* in = makeContigTensor(2);
```

Comment:
```suggestion
  auto* in = makeContigTensor(3);
```

Unfortunately, this change doesn't really make a behavioral difference, because postSingleCommunication won't look at the input and output TensorViews at all. However, a 3-D TensorView matches `input_tensor` being 3D. 

To fully match reality, `in->axis(0)` and `out->axis(1)` should in theory also be parallelized on DID. But, I'm fine with not doing that because (a) missing information (e.g. parallel types) is better that wrong information (e.g. 2 vs 3) and (b) tests are simpler without that.